Desktop_Interface/isobuffer.cpp

```diff
@@ -2,391 +2,364 @@
-    samplesInPast -= x0;
-    if(back < samplesInPast){
-        return -1; //too hard, not really important
-    }
-    short vbot_s = inverseSampleConvert(vbot, 2048, 0);
-    qDebug() << "vbot_s (x1) = " << vbot_s;
-
-    int num_found = 0;
-    for(int i=samplesInPast; i; i--){
-        short currentSample = buffer[back - i];
-        if(currentSample X1_X2_COMPARISON_CAP vbot_s){
-            num_found++;
-        } else num_found--;
-        if(num_found < 0){
-            num_found = 0;
-        }
-        if (num_found > NUM_SAMPLES_SEEKING_CAP){
-            return samplesInPast-i + x0;
-        }
-
-    }
-    return -1;
-}
+	if (self.back < samples + rbegin) return -1;
+
+	short sample = self.inverseSampleConvert(value, 2048, 0);
 
-int isoBuffer::cap_x2fromLast(double seconds, int x1, double vtop){
-    int samplesInPast = seconds * samplesPerSecond;
-    samplesInPast -= x1;
-    if(back < samplesInPast){
-        return -1; //too hard, not really important
-    }
-    short vtop_s = inverseSampleConvert(vtop, 2048, 0);
-    qDebug() << "vtop_s (x2) = " << vtop_s;
-
-    int num_found = 0;
-    for(int i=samplesInPast; i; i--){
-        short currentSample = buffer[back - i];
-        if(currentSample X1_X2_COMPARISON_CAP vtop_s){
-            num_found++;
-        } else num_found--;
-        if(num_found < 0){
-            num_found = 0;
-        }
-        if (num_found > NUM_SAMPLES_SEEKING_CAP){
-            return samplesInPast-i + x1;
-        }
-    }
-    return -1;
+	int found = 0;
+	for (int i = samples + rbegin; i--;)
+	{
+		short currentSample = self.bufferAt(i);
+		if (comp(currentSample, sample))
+			found = found + 1;
+		else
+			found = std::max(0, found-1);
+
+		if (found > target)
+			return samples - i;
+	}
+
+	return -1;
 }
 
-void isoBuffer::serialManage(double baudRate, int type, UartParity parity){
-    //Types:
-    // 0 - standard UART, no parity
-    // 1 - standard UART, with parity bit
-    // 100 - I2C
-    if(decoder == NULL){
-        decoder = new uartStyleDecoder(this);
-        connect(decoder, SIGNAL(wireDisconnected(int)), virtualParent, SLOT(serialNeedsDisabling(int)));
-    }
-    if(stopDecoding){
-        decoder->updateTimer->start(CONSOLE_UPDATE_TIMER_PERIOD);
-        stopDecoding = false;
-    }
-    decoder->setParityMode(parity);
-    decoder->serialDecode(baudRate);
+//For capacitance measurement. x0, x1 and x2 are all various time points used to find the RC coefficient.
+int isoBuffer::cap_x0fromLast(double seconds, double vbot)
+{
+	return capSample(*this, 0, NUM_SAMPLES_SEEKING_CAP, seconds, vbot, X0_COMP_FTOR);
 }
 
+int isoBuffer::cap_x1fromLast(double seconds, int x0, double vbot)
+{
+	return capSample(*this, -x0, NUM_SAMPLES_SEEKING_CAP, seconds, vbot, X1_X2_COMP_FTOR);
+}
 
+int isoBuffer::cap_x2fromLast(double seconds, int x1, double vtop)
+{
+	return capSample(*this, -x1, NUM_SAMPLES_SEEKING_CAP, seconds, vtop, X1_X2_COMP_FTOR);
+}
 
+// NOTE: type appears to be unused
```

Comment:
Yep, that's correct.  Type is unused and should be removed.
This function was originally going to manage i2c as well as rs-232 but that did not end up being the case.

Reply:
I removed it now